nbgrader/apps/zipcollectapp.py

```diff
@@ -0,0 +1,492 @@
+import os
+import six
+import sys
+import shutil
+import datetime
+
+from dateutil.tz import gettz
+from textwrap import dedent
+from traitlets import Bool, Instance, Type, Unicode
+from traitlets.config.application import catch_config_error, default
+
+from .baseapp import NbGrader
+
+from ..plugins import BasePlugin, ExtractorPlugin, FileNameCollectorPlugin
+from ..utils import check_directory, rmtree, parse_utc
+from ..utils import find_all_notebooks
+
+aliases = {
+    'log-level': 'Application.log_level',
+    'extractor': 'ZipCollectApp.extractor_plugin',
+    'collector': 'ZipCollectApp.collector_plugin',
+    'zip_ext': 'ExtractorPlugin.zip_ext',
+}
+flags = {
+    'debug': (
+        {'Application': {'log_level': 'DEBUG'}},
+        "set log level to DEBUG (maximize logging output)"
+    ),
+    'force': (
+        {
+            'ZipCollectApp': {'force': True},
+            'ExtractorPlugin': {'force': True}
+        },
+        "Force overwrite of existing files."
+    ),
+    'strict': (
+        {'ZipCollectApp': {'strict': True}},
+        "Skip submitted notebooks with invalid names."
+    ),
+}
+
+
+class ZipCollectApp(NbGrader):
+
+    name = u'nbgrader-zip-collect'
+    description = u'Collect assignments from archives (zip files).'
+
+    aliases = aliases
+    flags = flags
+
+    examples = """
+        Collect assignment submissions from files and/or archives (zip) files
+        manually downloaded from a LMS. For the usage of instructors.
+
+        This command is run from the top-level nbgrader folder. In order to
+        facilitate the collect process, nbgrader places some constraints on how
+        the manually downloaded archive (zip) files must be structured. By
+        default, the directory structure must look like this:
+
+            {downloaded}/{assignment_id}/{collect_step}
+
+        where `downloaded` is the main directory, `assignment_id` is the name
+        of the assignment and `collect_step` is the step in the collect
+        process.
+
+        Manually downloaded assignment submissions files and/or archives (zip)
+        files must be placed in the `archive_directory`:
+
+            {downloaded}/{assignment_id}/{archive_directory}
+
+        It is expected that the instructor has already created this directory
+        and placed the downloaded assignment submissions files and/or archives
+        (zip) files in this directory.
+
+        Archive (zip) files in the `archive_directory` will be extracted, and
+        any non-archive files will be copied, to the `extracted_directory`:
+
+            {downloaded}/{assignment_id}/{extracted_directory}
+
+        After which the files in the `extracted_directory` will be collected
+        and copied into the students `submitted_directory`:
+
+            {submitted_directory}/{student_id}/{assignment_id}/{notebook_id}.ipynb
+
+        By default the collection of files in the `extracted_directory` is
```

Comment:
I think you're still missing a reference to `FileNameProcessor` (and `--processor`) in the paragraph immediately after this one 😕 

I don't know what text editor you use, but if you happen to use Sublime, you can right click a folder in the left-hand file browser and select "Find in folder" to search all files within a folder. It's super useful, and a lot less painful than grep. grep in this particular case isn't too bad though:

```
grep -R FileNameProcessor .
``` 

and 

```
grep -R "\-\-processor" .
```

(both of those only reveal this one instance, so once that's fixed I think it should be good to go 😄 )